### PR TITLE
dts: Use separate compatibles for Nordic TWI/TWIM/TWIS and SPI/SPIM/SPIS peripherals

### PR DESCRIPTION
**Pull request by anangl**
**dts: Use separate compatibles for Nordic TWI/TWIM/TWIS peripherals**

This commit introduces separate "compatible" strings for dts nodes
representing different types of Nordic TWI peripherals. Previously
"nordic,nrf-i2c" was used for both TWI and TWIM, and TWIS was not
supported.

Quite a few files need to be touched by this commit but the changes can
be divided into groups of related or very similar ones, distinguishable
by the initial part of the path to the modified file:

* dts/bindings/i2c/
  new bindings for "nordic,nrf-twim" and "nordic,nrf-twis" are added
  and the one for "nordic,nrf-i2s" is renamed to "nordic,nrf-twi",
  common fields for all these bindings are extracted to a shared file

* dts/arm/nordic/
  "compatible" properties in i2cX nodes are updated (when there is no
  choice as only one type of TWI peripheral is available) or replaced
  with a comment pointing out that the proper type of peripheral needs
  to be picked at some upper layer

* drivers/i2c/
  both flavors of i2c_nrfx drivers are updated with the new names of
  macros generated from dts

* boards/
  all i2cX nodes in dts files for boards equipped with an nRF chip are
  updated with the proper "compatible" property, according to the type
  of TWI peripheral that is currently selected for the board by the
  corresponding Kconfig choice option (I2C_x_NRF_TWI*)

**dts: Use separate compatibles for Nordic SPI/SPIM/SPIS peripherals**

The set of changes done for SPI nodes is very similar. See the second commit message for details.

### CODE REVIEW

File: dts/bindings/spi/nordic,nrf-spim.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2019 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+title: Nordic nRF Family SPIM (SPI master with EasyDMA) node
```
Comment (ulfalizer) [inline dts/bindings/spi/nordic,nrf-spim.yaml]: Might be better to have something like

```
title: Nordic nRF family SPIM
description: Nordic nRF family SPIM (SPI master with EasyDMA) node
```

The `description` is what shows up in the comment in `generated_dts_board.h`. `title` is currently unused.

Comment (anangl) [reply dts/bindings/spi/nordic,nrf-spim.yaml]: Sure. Will switch those.

**Review by galak — APPROVED** [review]

**Comment by anangl** [comment]
Changed titles and descriptions in bindings as suggested by @ulfalizer.
Corrected the multi-line comments in .dtsi files to prevent checkpatch warnings.

File: dts/bindings/spi/nordic,nrf-spim.yaml (hunk 1 above)
Comment (anangl) [reply dts/bindings/spi/nordic,nrf-spim.yaml]: Done.

**Comment by anangl** [comment]
Rebased on master.

**Comment by anangl** [comment]
Added `compatible` properties in i2c and spi nodes for `actinius_icarus` board that was recently added to master.

**Review by Mierunski — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]
tested on reel board, LGTM

**Pull request merged**